lib/ansible/plugins/connection/psrp.py

```diff
@@ -579,8 +586,15 @@ def _build_kwargs(self):
         self._become_pass = self._play_context.become_pass
 
         self._psrp_host = self.get_option('remote_addr')
+
         self._psrp_user = self.get_option('remote_user')
-        self._psrp_pass = self._play_context.password
+        if self._psrp_user is not None:
+            # In case of an in-line vault string we want to convert to a text string
+            self._psrp_user = to_text(self._psrp_user)
```

Comment:
Seems like *maybe* `get_option` should handle this internally, just like with templating... @bcoca, thoughts?

Reply:
type: string should do that

Reply:
@bcoca unfortunately no, it fails with

`Invalid type for configuration option plugin_type: connection plugin: winrm setting: remote_user : Invalid type provided for "string": option value`

Sounds like `type: str` should be doing `to_text()` internally though.

Reply:
Alternative PR for the above is https://github.com/ansible/ansible/pull/65023